src/main/flight/imu.c

```diff
@@ -201,11 +188,33 @@ void acc_calc(uint32_t deltaT)
 /*
 * Baseflight calculation by Luggi09 originates from arducopter
 * ============================================================
+* This function turns a vector which is (usually) the direction
+* of magnetic flux in the coordinate system of the craft into
+* a compass heading in degrees, clockwise away from north. Note
+* that the magnetic flux is not parrelell with the vector towards
+* magnetic north it's self but rather is parrelell with the ground
+* when near the equator.
```

Comment:
Thanks very much! I didn't really have time to look into it in great depth, but with your added comments I'll take some time to make a really nice explanation :)

What I was trying to emphasize is that the B field is from a dipole not a point.